sql-ballerina/tests/call-procedures-test.bal

```diff
@@ -0,0 +1,393 @@
+
+    test:assertEquals(paraVarchar.get(string), "test0", "2nd out parameter of procedure did not match.");
+    test:assertEquals(paraCharmax.get(string), "test1     ", "3rd out parameter of procedure did not match.");
+    test:assertEquals(paraChar.get(string), "a", "4th out parameter of procedure did not match.");
+    test:assertEquals(paraCharactermax.get(string), "test2     ", "5th out parameter of procedure did not match.");
+    test:assertEquals(paraCharacter.get(string), "b", "6th out parameter of procedure did not match.");
+    test:assertEquals(paraNvarcharmax.get(string), "test3", "7th out parameter of procedure did not match.");
+}
+
+
+@test:Config {
+    groups: ["procedures"],
+    dependsOn: ["testCallWithStringTypesInoutParams","testCreateProcedures5"]
+}
+function testCallWithNumericTypesInoutParams() {
+    MockClient dbClient = checkpanic new (url = proceduresDB, user = user, password = password);
+    decimal paraInDecimalVal= -1234.56;
+
+    IntegerValue paraID = new(1);
+    InOutParameter paraInt = new(-2147483647);
+    InOutParameter paraBigInt = new(-9223372036854774807);
+    InOutParameter paraSmallInt = new(-32767);
+    InOutParameter paraTinyInt = new(-127);
+    InOutParameter paraBit = new(false);
+    InOutParameter paraDecimal = new(paraInDecimalVal);
+    InOutParameter paraNumeric = new(paraInDecimalVal);
+    InOutParameter paraFloat = new(-1234.56);
+    InOutParameter paraReal = new(-1234.56);
+    InOutParameter paraDouble = new(-1234.56);
+
+    var ret = checkpanic dbClient->call(
+        `{call SelectNumericDataWithInoutParams(${paraID}, ${paraInt}, ${paraBigInt}, ${paraSmallInt}, ${paraTinyInt}, ${paraBit}, ${paraDecimal}, ${paraNumeric}, ${paraFloat}, ${paraReal}, ${paraDouble})}`);
+    checkpanic dbClient.close();
+
+    decimal paraDecimalVal= 1234.56;
+
+    test:assertEquals(paraInt.get(int), 2147483647, "2nd out parameter of procedure did not match.");
+    test:assertEquals(paraBigInt.get(int), 9223372036854774807, "3rd out parameter of procedure did not match.");
+    test:assertEquals(paraSmallInt.get(int), 32767, "4th out parameter of procedure did not match.");
+    test:assertEquals(paraTinyInt.get(int), 127, "5th out parameter of procedure did not match.");
+    test:assertEquals(paraBit.get(boolean), true, "6th out parameter of procedure did not match.");
+    test:assertEquals(paraDecimal.get(decimal), paraDecimalVal, "7th out parameter of procedure did not match.");
+    test:assertEquals(paraNumeric.get(decimal), paraDecimalVal, "8th out parameter of procedure did not match.");
+    test:assertTrue((checkpanic paraFloat.get(float)) > 1234.0, "9th out parameter of procedure did not match.");
+    test:assertTrue((checkpanic paraReal.get(float)) > 1234.0, "10th out parameter of procedure did not match.");
+    test:assertEquals(paraDouble.get(float), 1234.56, "11th out parameter of procedure did not match.");
+}
+
+
+
+
+
+@test:Config {
+    groups: ["procedures"]
```

Comment:
If all test cases under `procedures` depend on a single procedure. Better use dependsOn to verify the tests are run in order. This will reduce if there is any intermittent test failures.

Reply:
Ok. I will use it

Reply:
I added dependsOn parameter. commit id is ```76d0e08c47bcd7120a490db97680c560671e913a```